src/prefect/server/api/server.py

```diff
@@ -499,7 +499,9 @@ def create_app(
             and services will be disabled.
     """
     settings = settings or prefect.settings.get_current_settings()
-    cache_key = (settings, ephemeral)
+    env_variables = settings.to_environment_variables()
+    settings_key = hash(tuple((key, value) for key, value in env_variables.items()))
```

Comment:
This feels like something that maybe our settings should handle. What do you think about adding a method to the settings that's like:

```python
def hash_key(self) -> str:
    env_variables = self.to_environment_variables()
    return str(hash(tuple((key, value) for key, value in env_variables.items())))
```

And then we can call that here: `cache_key = (settings.hash_key(), ephemeral)`

Reply:
Thanks for the feedback, I'll try to get to this in the next 48 hours.

Reply:
Updated PR.